addons/website/static/src/components/dialog/seo.js

```diff
@@ -369,13 +591,169 @@ class TitleDescription extends Component {
     }
 }
 
+export class SeoChecks extends Component {
+    static template = "website.SeoChecks";
+    static components = {
+        CheckBox,
+    };
+    static props = {};
+
+    setup() {
+        this.website = useService("website");
+        this.seoContext = useState(seoContext);
+        this.state = useState({
+            altAttributes: this.getAltAttributes(),
+            checkingLinks: false,
+            checkingLink: false,
+            checkedLinks: false,
+            counterLinks: 0,
+            totalLinks: 0,
+            headingsScan: this.getHeadingsScan(),
+        });
+        this.imgUpdated = this.imgUpdated.bind(this);
+    }
+
+    imgUpdated(img) {
+        img.updated = true;
+        this.seoContext.updatedAlts = this.state.altAttributes.filter(img => img.updated);
+    }
+
+    getHeadingsScan() {
+        const ret = {
+            missingH1: false,
+            multipleH1: false,
+            misplacedH1: false,
+        };
+        const allHeadingsEls = Array.from(
+            this.website.pageDocument.documentElement.querySelectorAll("#wrap h1,h2,h3,h4,h5,h6")
+        );
```

Comment:
It's strange to use this selector: `#wrap h1,h2,h3,h4,h5,h6` which will return the `h2~h6` outside the `#wrap`. E.g., If you have a page with a Mega Menu (the one with `h4` titles), you'll always get a "misplaced h1" warning even with `h1` as the first heading in the main content area :grimacing: